swift-sdk/Internal/InternalIterableAPI.swift

```diff
@@ -271,8 +275,11 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
                     mergeNestedObjects: Bool,
                     onSuccess: OnSuccessHandler? = nil,
                     onFailure: OnFailureHandler? = nil) -> Pending<SendRequestValue, SendRequestError> {
-        if !isEitherUserIdOrEmailSet() && config.enableAnonTracking && localStorage.userIdAnnon == nil{
-            anonymousUserManager.trackAnonUpdateUser(dataFields)
+        if !isEitherUserIdOrEmailSet() && localStorage.userIdAnnon == nil {
+            if config.enableAnonTracking {
+                anonymousUserManager.trackAnonUpdateUser(dataFields)
+            }
+            return rejectWithInitializationError(onFailure: onFailure)
```

Comment:
Remove the rejectWithInitializationError

Reply:
if you notice we need to return the result of type Pending<SendRequestValue, SendRequestError> and that is why we cannot remove rejectWithInitializationError.